lib/symbol_table.rb

```diff
@@ -33,6 +33,10 @@ def start_subroutine
     @subroutine_symbol_table = {}
   end
 
+  def defined?(name)
```

Comment:
This helper’s a good idea, but `#defined?` is a slightly misleading name for it because `defined?` is a [Ruby keyword](https://docs.ruby-lang.org/en/3.1/doc/syntax/miscellaneous_rdoc.html#label-defined-3F) for (kind of) looking into the _Ruby_ symbol table:

```
>> foo, @bar = 1, 2
=> [1, 2]
>> defined?(foo)
=> "local-variable"
>> defined?(@bar)
=> "instance-variable"
>> defined?(baz)
=> nil
```

Something like `#include?` might be less surprising.